test/frames/frame_request_controller_test.rb

```diff
@@ -35,6 +35,22 @@ class Turbo::FrameRequestControllerTest < ActionDispatch::IntegrationTest
     assert_not_equal etag_with_frame, etag_without_frame
   end
 
+  test "frame requests keep the flash" do
+    message = Message.create!
+
+    post messages_path
+    assert_equal @request.flash[:notice], 'Message was successfully created.'
+
+    get messages_path, headers: { "Turbo-Frame" => "true" }
+    assert_equal @request.flash[:notice], 'Message was successfully created.'
+
+    get messages_path
+    assert_equal @request.flash[:notice], 'Message was successfully created.'
+
+    get messages_path
+    assert_nil @request.flash[:notice]
```

Comment:
Hmm, the message was required in my initial change as I was following the redirect to `message_url(id: 1)`.
In the current implementation it isn't required. I think this could use a little clean up.